tests/scenario/test_machine_charm/test_models.py

```diff
@@ -45,6 +45,7 @@ def test_cos_agent_provider_unit_data_dashboard_serialization():
         "metrics_scrape_jobs": [],
         "log_slots": [],
         "subordinate": True,
+        "tracing_protocols": None,
```

Comment:
is it a required field now? That will break BC.

Reply:
This one is an Optional:
```
    # Requested tracing protocols.
    tracing_protocols: Optional[List[str]] = None
```

however this makes me realize that with the introduction of contents in the databag on the other side (list of published receivers) we need to think on what to do with `cos-proxy` that won't have support for tracing and also uses cos-agent. Or, maybe, should it have the support as well?

Reply:
Ok, I'm wrong about `cos-proxy`: it uses cos-agent library, yes, but as a `Provider` - so it doesn't act in the same way as grafana-agent. Probably we'll be okay with that.

Reply:
I can't follow this right now but if you're convinced I'm convinced :)

Reply:
Tried relating zookeeper with an older cos-agent version to grafana-agent with these model changes: new fields are just ignored so it looks like it's fine.